kview/src/main/java/dev/ikm/komet/kview/controls/skin/KLComponentControlSkin.java

```diff
@@ -216,34 +232,141 @@ private void setupDragNDrop() {
                 }
 
                 try {
+                    int nid = extractNid(event);
                     LOG.info("publicId: {}", dragboard.getString());
-                    if (event.getGestureSource() instanceof Node source &&
-                            source.getUserData() instanceof DragAndDropInfo dropInfo &&
-                            dropInfo.publicId() != null
-                    ) { // TODO: should this be needed? shouldn't we get PublicId from dragboard content?
-                        int nid = EntityService.get().nidForPublicId(dropInfo.publicId());
-                        EntityProxy entity = EntityProxy.make(nid);
+                    if (nid != Integer.MIN_VALUE) {  //
```

Comment:
We stopped using Integer.MIN_VALUE to denote an empty component control. 
There's a method called isEmpty() that can be used, here would be `getSkinable().isEmpty()`

Reply:
Please examine the method extractNid(event). if not found it will use this reserved integer to denote not found.
All this code should be refactored to keep much of this database specific code out of the skin. But this was much needed as the user will want to drag components from different ui controls as before.